pkg/controller/mongodb/mongodb_status_options.go

```diff
@@ -99,18 +80,35 @@ type messageOption struct {
 func (m messageOption) ApplyOption(mdb *mdbv1.MongoDB) {
 	mdb.Status.Message = m.message.messageString
 	if m.message.severityLevel == Error {
-		zap.S().Error(m.message)
+		zap.S().Error(m.message.messageString)
 	}
 	if m.message.severityLevel == Warn {
-		zap.S().Warn(m.message)
+		zap.S().Warn(m.message.messageString)
 	}
 	if m.message.severityLevel == Info {
-		zap.S().Info(m.message)
+		zap.S().Info(m.message.messageString)
+	}
+	if m.message.severityLevel == Debug {
+		zap.S().Debug(m.message.messageString)
 	}
 }
 
 func (m messageOption) GetResult() (reconcile.Result, error) {
-	return okResult()
+	return result.OK()
+}
+
+func (o *optionBuilder) withAutomationConfigMembers(members int) *optionBuilder {
```

Comment:
Although this API seems super ergonomic, it is not :)

There's no way (easy way) to see **which attribute on the `status` this function will end up modifying**.

If I call `withAutomationConfigMembers(n)`, will it end up doing `status.currentStatefulSetReplicas = n` or `status.currentReplicaSetMembers = n`?

Reply:
From a developer's perspective, it would be as easy as to explicitly describe which `status` attribute this function is supposed to change.

```golang
// withAutomationConfigMembers sets `status.someImportantAttribute`.
func withAutomationConfigMembers () {}